src/domain/creator/generateNickname.ts

```diff
@@ -0,0 +1,37 @@
+
+import { TooManySimilarNickNames } from './errors';
+import retrieveByNickName from './data/retrieveByNickName';
+import retrieveByStartNickName from './data/retrieveByStartNickName';
+
+export default async function generateNickname(nickname: string): Promise<string>
+{
+    const strippedName: string = nickname.trim();
+    const noSpacesNickName: string = strippedName.replaceAll(' ', '');
+    const cleanNickName: string = noSpacesNickName.replaceAll('_', '');
+
+    const existingData = await retrieveByNickName(cleanNickName);
+
+    if (existingData === undefined)
+    {
+        return cleanNickName;
+    }
+
+    const foundNickName = await retrieveByStartNickName(`${existingData.nickName}_`);
```

Comment:
The const name `foundNickName` doesn't fit the result of function anymore because it now retrieves a creator object instead of a string.